src/jesse_json_path.erl

```diff
@@ -302,3 +302,16 @@ parse_json_pointer_token(Token) ->
                , {<<"~1">>, <<"/">>}
                ]
              ).
+
+%% @private
+hex_decode([$%, Hex1, Hex2 | Rest]) ->
```

Comment:
> decode is replaced by module private hex_decode function which is based on http_uri:decode code, because I can not find alternative in uri_string module.

can you inline your comment from the PR? I believe it's the very first thought of everyone, but I also cannot find a built-in function for it

Reply:
PS: I haven't tested this, but could it be that when using uri_string:parse, we don't have to bother with hex encodings?

Reply:
Just tested, `%00` is left as is :crying_cat_face: 
```
6> uri_string:parse(<<"http://example.com/baz%00">>).
#{host => <<"example.com">>,path => <<"/baz%00">>,
  scheme => <<"http">>}
```

Reply:
Some context: https://github.com/erlang/otp/pull/2492